src/components/Search/Search.tsx

```diff
@@ -0,0 +1,51 @@
+import s from '../Header/Header.module.css';
+import React, { useEffect, useRef, useState } from 'react';
+
+const Search = () => {
+  const [value, setValue] = useState<string>('');
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', componentCleanup);
+    const storageValue = localStorage.getItem('search');
```

Comment:
const enum LocalStorageKeys  { Search = 'search' }

...
storageValue.getItem(LocalStorageKeys.Search) we can try